### PR TITLE
PositionStreams are now canceled when Location Services are disabled

### PR DESCRIPTION
**Pull request by florissmit1**
### :sparkles: What kind of change does this PR introduce? (Bug fix, feature, docs update...)
Example app update

### :arrow_heading_down: What is the current behavior?
Currently, when Location Services are disabled on the user's mobile device the PositionStream is canceled and cannot be continued.

### :new: What is the new behavior (if this is a feature change)?
By making use of the Location Service Stream you can prevent this problem. The example app now makes use of this.

### :boom: Does this PR introduce a breaking change?
No

### :bug: Recommendations for testing
Use the example application

### :memo: Links to relevant issues/docs
Does not apply

### :thinking: Checklist before submitting

- [ ] I made sure all projects build.
- [ ] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy](https://dart.dev/tools/pub/versioning).
- [ ] I updated CHANGELOG.md to add a description of the change.
- [ ] I followed the style guide lines ([code style guide](https://github.com/Baseflow/flutter-geolocator/blob/master/CONTRIBUTING.md)).
- [ ] I updated the relevant documentation.
- [ ] I rebased onto current `master`.

### CODE REVIEW

File: geolocator/example/ios/Flutter/AppFrameworkInfo.plist [hunk 1]
```diff
@@ -21,6 +21,6 @@
   <key>CFBundleVersion</key>
   <string>1.0</string>
   <key>MinimumOSVersion</key>
-  <string>8.0</string>
+  <string>9.0</string>
```
Comment (mvanbeusekom) [inline geolocator/example/ios/Flutter/AppFrameworkInfo.plist:24]: Why this change? Doesn't seem to be necessary, maybe you can rollback the changes in this file (together with the changes in `project.pbxproj`)

File: geolocator/example/ios/Runner.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -332,7 +332,7 @@
 				GCC_WARN_UNINITIALIZED_AUTOS = YES_AGGRESSIVE;
 				GCC_WARN_UNUSED_FUNCTION = YES;
 				GCC_WARN_UNUSED_VARIABLE = YES;
-				IPHONEOS_DEPLOYMENT_TARGET = 8.0;
+				IPHONEOS_DEPLOYMENT_TARGET = 9.0;
```
Comment (mvanbeusekom) [inline geolocator/example/ios/Runner.xcodeproj/project.pbxproj:335]: Why is it necessary to move to deployment target 9.0?

File: geolocator/example/ios/Runner.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -414,7 +414,7 @@
...
```
Comment (mvanbeusekom) [inline geolocator/example/ios/Runner.xcodeproj/project.pbxproj:417]: Same

File: geolocator/example/ios/Runner.xcodeproj/project.pbxproj [hunk 4]
```diff
@@ -463,7 +463,7 @@
...
```
Comment (mvanbeusekom) [inline geolocator/example/ios/Runner.xcodeproj/project.pbxproj:466]: Same

File: geolocator/example/lib/main.dart [hunk 5]
```diff
@@ -154,7 +155,14 @@ class _GeolocatorWidgetState extends State<GeolocatorWidget> {
                             _positionStreamSubscription!.isPaused)
                         ? const Icon(Icons.play_arrow)
                         : const Icon(Icons.pause),
-                    onPressed: _toggleListening,
+                    onPressed: () {
+                      if (positionStreamStarted) {
+                        positionStreamStarted = false;
+                      } else {
+                        positionStreamStarted = true;
+                      }
```
Comment (mvanbeusekom) [inline geolocator/example/lib/main.dart:163]: You can shorten this a bit by using:
```suggestion
                      positionStreamStarted = !positionStreamStarted;
```

File: geolocator_apple/example/ios/Flutter/AppFrameworkInfo.plist [hunk 6]
```diff
@@ -21,6 +21,6 @@
   <key>CFBundleVersion</key>
   <string>1.0</string>
   <key>MinimumOSVersion</key>
-  <string>8.0</string>
+  <string>9.0</string>
```
Comment (mvanbeusekom) [inline geolocator_apple/example/ios/Flutter/AppFrameworkInfo.plist:24]: Same as before, why the need to update deployment target?

File: geolocator_apple/example/ios/Runner.xcodeproj/project.pbxproj [hunk 7]
```diff
@@ -332,7 +332,7 @@
 				GCC_WARN_UNINITIALIZED_AUTOS = YES_AGGRESSIVE;
 				GCC_WARN_UNUSED_FUNCTION = YES;
 				GCC_WARN_UNUSED_VARIABLE = YES;
-				IPHONEOS_DEPLOYMENT_TARGET = 8.0;
+				IPHONEOS_DEPLOYMENT_TARGET = 9.0;
```
Comment (mvanbeusekom) [inline geolocator_apple/example/ios/Runner.xcodeproj/project.pbxproj:335]: Same

File: geolocator_apple/example/ios/Runner.xcodeproj/project.pbxproj [hunk 8]
```diff
@@ -414,7 +414,7 @@
...
```
Comment (mvanbeusekom) [inline geolocator_apple/example/ios/Runner.xcodeproj/project.pbxproj:417]: Same

File: geolocator_apple/example/ios/Runner.xcodeproj/project.pbxproj [hunk 9]
```diff
@@ -463,7 +463,7 @@
...
```
Comment (mvanbeusekom) [inline geolocator_apple/example/ios/Runner.xcodeproj/project.pbxproj:466]: Same

**Review by mvanbeusekom — CHANGES_REQUESTED** [review]
Have a question about changing the iOS deployment target and a small code suggestion.

File: geolocator/example/lib/main.dart (hunk 5 above)
Comment (florissmit1) [reply geolocator/example/lib/main.dart:163]: Thanks, I thought it was 'cleaner' to make use of a if statement in this case.

**Comment by mvanbeusekom** [comment]
@florissmit1 looks good, only needs to update the pubspec.yaml and changelog.md with a new version (patch).

**Review by mvanbeusekom — APPROVED** [review]
LGTM

**Pull request merged**